setup.py

```diff
@@ -100,7 +100,11 @@ def get_version():
     for line in version_file:
         if line.startswith("__version__"):
             version_tuple = ast.literal_eval(line.split(" = ")[1])
-            return ".".join(map(str, version_tuple))
+            version_str = ".".join(map(str, version_tuple))
+            local_version = os.getenv("ELASTIC_CI_LOCAL_VERSION")
+            if local_version:
+                return f"{version_str}-dev{local_version}"
```

Comment:
as discussed in https://github.com/elastic/apm-agent-python/pull/2000#issuecomment-1988917372